src/test/java/com/wikia/webdriver/testcases/renametool/RenameToolTests.java

```diff
@@ -0,0 +1,242 @@
+
+    new ArticleContent().push("aaa", String.format("User:%s", user.getUsername()));
+
+    UserProfilePage userProfilePage = new UserProfilePage()
+        .open(user.getUsername());
+
+    SpecialRenameUserPage renameUserPage = new SpecialRenameUserPage(driver)
+        .open()
+        .fillFormData("NewUser Nąmę" + timestamp, "NewUser Nąmę" + timestamp, user.getPassword())
+        .agreeToTermsAndConditions()
+        .submitChange();
+    new ConfirmationModalPage(driver).accept();
+    Assertion
+        .assertEquals(renameUserPage.getSuccessBoxMessage(), "Rename process is in progress. The "
+                                                             + "rest will be done in background. You will be notified via e-mail when it is completed.");
+  }
+
+  @Test(groups = {"renameTool_01"})
+  public void NewUserCreateEditMessageWallAndRenameDone() {
+    Credentials credentials = new Credentials();
+    String timestamp = Long.toString(DateTime.now().getMillis());
+    SignUpUser
+        user =
+        new SignUpUser("QARename Usęr" + timestamp, credentials.email, "aaaa",
+                       LocalDate.of(1993, 3, 19));
+    UserRegistration.registerUserEmailConfirmed(user);
+
+    new WikiBasePageObject().loginAs(user.getUsername(), user.getPassword(), wikiURL);
+
+    MessageWall wall = new MessageWall().open(user.getUsername());
+    MiniEditorComponentObject mini = wall.triggerMessageArea();
+    String message = PageContent.MESSAGE_WALL_MESSAGE_PREFIX + wall.getTimeStamp();
+    String title = PageContent.MESSAGE_WALL_TITLE_PREFIX + wall.getTimeStamp();
+    mini.switchAndWrite(message);
+    wall.setTitle(title);
+    wall.submit();
+    wall.verifyMessageText(title, message, user.getUsername());
+    wall.triggerEditMessageArea();
+    String messageEdit = PageContent.MESSAGE_WALL_MESSAGE_EDIT_PREFIX + wall.getTimeStamp();
+    mini.switchAndEditMessageWall(messageEdit);
+    wall.submitEdition();
+    wall.verifyMessageEditText(title, messageEdit, user.getUsername());
+
+    String newName = "NewUser Nąmę" + timestamp;
+    SpecialRenameUserPage renameUserPage = new SpecialRenameUserPage(driver)
+        .open()
+        .fillFormData(newName, newName, user.getPassword())
+        .agreeToTermsAndConditions()
+        .submitChange();
+    new ConfirmationModalPage(driver).accept();
+    Assertion
+        .assertEquals(renameUserPage.getSuccessBoxMessage(), "Rename process is in progress. The "
+                                                             + "rest will be done in background. You will be notified via e-mail when it is completed.");
+    wall.open(newName);
+    String encodedUrl = renameUserPage.encodeToURL(newName);
+
+    String expectedWallUrl = String.format("%s/wiki/Message_Wall:%s", urlBuilder.getUrlForWiki(),
+                                           encodedUrl);
+
+    Assertion.assertEquals(driver.getCurrentUrl(), expectedWallUrl);
+    wall.verifyMessageEditTextRenameDone(title, message, newName);
```

Comment:
would be sweet to have assertion here instead of in another layer of abstraction (inside method `verifyMessageEditTextRenameDone` )